packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -630,7 +630,20 @@ public boolean onActivityResult(
 
   private void handleChooseImageResult(int resultCode, Intent data) {
     if (resultCode == Activity.RESULT_OK && data != null) {
-      String path = fileUtils.getPathFromUri(activity, data.getData());
+      Uri uri = null;
+      // On several pre-Android 13 devices using Android Photo Picker, the Uri from getData() could be null.
+      if (data.getData() != null) {
+        uri = data.getData();
+      } else if (data.getClipData() != null && data.getClipData().getItemCount() == 1) {
+        uri = data.getClipData().getItemAt(0).getUri();
+
+        if (uri == null) {
```

Comment:
This should not be nested here; as written there is a path through this method where the URI couldn't be extracted either way, but no error is returned here and `getPathFromUri` is still called.

Reply:
(Which also means there's a missing unit test for that case.)